server/clip_server/executors/clip_onnx.py

```diff
@@ -114,37 +121,32 @@ async def rank(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
                     _txt_da.embeddings, axis=1, keepdims=True
                 )
 
-                # cosine similarity as logits
-                logit_scale = np.exp(self._logit_scale)
-                logits_per_image = logit_scale * np.matmul(
-                    image_features, text_features.T
-                )
-                logits_per_text = logits_per_image.T
-
-                def numpy_softmax(z):
-                    s = np.max(z, axis=1)
-                    s = s[:, np.newaxis]
-                    e_x = np.exp(z - s)
-                    div = np.sum(e_x, axis=1)
-                    div = div[:, np.newaxis]  # dito
-                    return e_x / div
+                # paired cosine similarity
+                scores_per_text = np.matmul(image_features, text_features.T)
+                scores_per_image = scores_per_text.T
 
                 if len(_img_da) == 1:
-                    probs = numpy_softmax(logits_per_image)[0]
+                    scores = scores_per_text
                 elif len(_txt_da) == 1:
-                    probs = numpy_softmax(logits_per_text)[0]
+                    scores = scores_per_image
+
+                if _score == 'probability':
+                    scores = numpy_softmax(self._logit_scale * scores)
+
+                # squeeze scores
+                scores = scores[0]
 
                 # drop embeddings
                 _img_da.embeddings = None
                 _txt_da.embeddings = None
 
-                for c, v in zip(_get(d), probs):
+                for c, v in zip(candidates, scores):
                     c.scores['clip_score'].value = v
```

Comment:
add `c.scores['clip_score'].op_name = score_format` to keep track the meaning of the score

Reply:
https://docarray.jina.ai/proto/?utm_source=docarray#docarray-NamedScoreProto